test/any/datastore/rate_limited_DAO-test.es6.js

```diff
@@ -45,28 +47,39 @@ describe('RateLimitedDAO', function() {
     rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 2}));
     rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 3}));
     jasmine.clock().tick(160);
-    rateLimitedDAO.delegate.select().then((arraySink) => {
-      expect(arraySink.a.map((Item) => Item.id).sort()).toEqual(
+    arrayDAO.select().then((arraySink) => {
+      expect(arraySink.a.map((Item) => Item.id)).toEqual(
         [1, 2, 3]);
         done();
     });
   });
-  it('handles more complex scenariou.', function(done) {
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 1}));
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 2}));
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 3}));
-    jasmine.clock().tick(70);
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 4}));
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 5}));
-    jasmine.clock().tick(150);
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 6}));
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 7}));
-    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 8}));
-    jasmine.clock().tick(300);
-    rateLimitedDAO.delegate.select().then((arraySink) => {
-      expect(arraySink.a.map((Item) => Item.id).sort()).toEqual(
-        [1, 2, 3, 4, 5, 6, 7, 8]);
-        done();
-    });
+  it('accepts puts after partial dequeue and empties queue correctly.',
+    function(done) {
+      rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 1}));
+      rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 2}));
+      rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 3}));
+      jasmine.clock().tick(70);
+      rateLimitedDAO.delegate.select().then((arraySink) => {
```

Comment:
Comparing this to line 50: We should either refer to `arrayDAO` or refer to `rateLimitedDAO.delegate` when doing `select()` for verification. Either's fine; let's just make it consistent.